### PR TITLE
Add image filtering by tag name

### LINKED ISSUE #6: Images by Tag
#### As a user I want to view all images associated with a tag.

__Story__:

Now that we have added the ability for images to be associated with tags, let
us provide the ability for users to filter images by a given tag. Nana, who
loves fast planes, will be delighted to see only images labeled with SR-71.

__Acceptance criteria__:
- [ ] When I click on a tag, I see a filtered list of only the images that have
  the tag I clicked on.
- [ ] The tag name, not its ID, is used to perform the filtering.

__Discussion topics__:
- [ ] Where in the code should we implement this functionality?
- [ ] What happens if a user tries to manually filter by a nonexistent tag?

__Dependencies__:
- Image Tags

### PR DESCRIPTION
**Pull request by aarontcote**
Closes #6

### CODE REVIEW

File: app/controllers/images_controller.rb [hunk 1]
```diff
@@ -1,6 +1,10 @@
 class ImagesController < ApplicationController
   def index
-    @images = Image.order('created_at DESC')
+    @images = if params[:tag]
```
Comment (brightmer) [inline app/controllers/images_controller.rb:3]: Discussion topics:
 What happens if a user tries to manually filter by a nonexistent tag?
 
 You could copy https://github.com/AppFolioOnboarding/image-sharer-aarontcote/blob/addImageTagFiltering/test/controllers/images/images_controller_index_test.rb#L24
 but use `get images_path(tag: 'tag_not_in_db')`

Comment (aarontcote) [reply app/controllers/images_controller.rb:3]: When searched by a tag that doesn't exist, you get an index page the shows the no images card, and I'll add testing to include those cases.

Comment (aarontcote) [reply app/controllers/images_controller.rb:3]: Updated, and ready for your review.

File: test/controllers/images/images_controller_index_test.rb [hunk 2]
```diff
@@ -21,13 +21,39 @@ class ImagesControllerIndexTest < ActionDispatch::IntegrationTest
...
     assert_response :ok
 
-    assert_select 'h3.text-center'
+    assert_select '.js-no-images'
+  end
+
+  test 'index should show no when tag is not used' do
+    setup_sample_image_data
+
+    get images_path(tag: TAG_NAME_INVALID)
```
Comment (brightmer) [inline test/controllers/images/images_controller_index_test.rb:53]: Thanks for adding this, one more super picky comment is that adding the const for the tags is a little overly abstract, for the tests. I (and others) prefer a more explicit test that doesn't require looking in another file to see what is being tested. But for this repository it is great. :shipit:

**Review by brightmer — APPROVED** [review]

**Pull request merged**